### PR TITLE
Adding the documentations for 5 new lint rules 

### PR DESCRIPTION
**Pull request by jianyexi**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [ ] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md [hunk 1]
```diff
@@ -2394,3 +2399,187 @@ The following would be invalid:
...
+
+### <a name="r4010" ></a>R4010 RequiredDefaultResponse
+
+**Category** : ARM Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response is defined without a default error response implementing,please add it. 
+
+**Description** : Per ARM Specs, Every operation must have a default error response implementing.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:2436]: “, **every** operation must have ... **implementation**”

File: documentation/openapi-authoring-automated-guidelines.md [hunk 2]
```diff
@@ -2394,3 +2399,187 @@ The following would be invalid:
...
+
+**CreatedAt**: May 21, 2020
+
+**LastModifiedAt**: May 21, 2020
+
+**How to fix the violation**: For each operation response add a default error response implementing:
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:2442]: For each operation response**,** please add a default ... **implementation**.

File: documentation/openapi-authoring-automated-guidelines.md [hunk 3]
```diff
@@ -2394,3 +2399,187 @@ The following would be invalid:
...
+
+### <a name="r4011" ></a>R4011 DeleteOperationResponses
+
+**Category** : ARM Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The delete operation is defined without a 200 or 204 error response implementing,please add it. 
+
+**Description** : Per ARM Specs, All DELETE methods must have responses code implementing: 200, 204.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:2468]: “, **all** DELETE methods ... **implementation**.”

Comment (josefree) [reply documentation/openapi-authoring-automated-guidelines.md:2468]: implementing should be implementation.  please search and fix all other similar typos.  thanks.

**Review by josefree — CHANGES_REQUESTED** [review]

File: documentation/openapi-authoring-automated-guidelines.md (hunk 2 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: offline communicated

**Review by josefree — APPROVED** [review]

**Review by akning-ms — APPROVED** [review]

**Comment by bradenwatkins** [comment]
Question: For linting rule R4011 it says, "Per ARM Specs, all DELETE methods must have responses code implementation: 200, 204."

Can you share a link to the mentioned "ARM Specs"?  It would be helpful to have a pointer back to the spec what is driving each rule.

**Comment by jianyexi** [comment]
> Question: For linting rule R4011 it says, "Per ARM Specs, all DELETE methods must have responses code implementation: 200, 204."
> 
> Can you share a link to the mentioned "ARM Specs"? It would be helpful to have a pointer back to the spec what is driving each rule.

Good point,  the link is here https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource

**Pull request merged**